### PR TITLE
Added cloud_pipline for E4S on Amazon Linux

### PR DESCRIPTION
**Pull request by bollig**

### CODE REVIEW

**Comment by bollig** [comment]
Still a draft. Reclassing to get gitlab-ci to execute.

**Comment by bollig** [comment]
depends on patch for https://github.com/spack/spack/issues/28830 to be applied in gitlab-runner or .gitlab-ci.yaml.

**Comment by bollig** [comment]
Logging issue for others to find later. The gitlab CI showed: 
image


```
Found errors in your .gitlab-ci.yml:
(specs) cuda/uriwfid 11.6.1 gcc@7.3.1 linux-amzn2-x86_64 AWS Packages job: undefined need: (specs) libxml2/g33bzmo 2.9.12 gcc@7.3.1 linux-amzn2-x86_64 AWS Packages
You can also test your .gitlab-ci.yml in [CI Lint](https://gitlab.spack.io/spack/spack/-/ci/editor?tab=LINT_TAB)
```
This error is another way of saying my catch-all match in `gitlab-ci:mappings` was for `os=ubuntu18.04` rather than `os=amzn2`. The OS needs to match the gitlab runner image from the generate phase.

**Comment by bollig** [comment]
image

Gitlab-CI is configured with Docker image pull `if-not-present` (images are pulled only if they don't exist in the host cache). For long-running runners, this means `:latest` yesterday is considered the same as `:latest` today. Considering whether it would make sense to support [multiple pull policies](https://docs.gitlab.com/runner/executors/docker.html#using-multiple-pull-policies), but in the meantime, specifying the full hash for the image (i.e., `image: { "name": "ghcr.io/spack/e4s-amazonlinux-2:latest@sha256:81721f939f15adace398dcf4a841717a9495b6cabaf8787914e3794c5ec3cbd7", "entrypoint": [""] }`) forces a pull on the next pipeline run across all runners.

**Comment by bollig** [comment]
Ready for review. Note this is a paired down version of E4s Pipeline. 

Disabled on ARM due to x86-specific instructions, lack of SVE, or cuda failures: ginkgo, hpx +cuda, hpx, llvm, petsc +cuda, phist, slepc +cuda, variorum, 
Disabled on x86_64 due to cuda issues: petsc +cuda, slepc +cuda

File: share/spack/gitlab/cloud_pipelines/stacks/aws-e4s-aarch64/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+    - - $default_specs
+    - - $compiler
+    - - $target
+
+  - matrix:
+    - - $cuda_specs
+    - - $compiler
+    - - $target
+
+  mirrors: { "mirror": "s3://spack-binaries/aws-e4s" }
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/aws-e4s-aarch64/spack.yaml]: Not incorrect, but I just prefer to match the stack name here.

File: share/spack/gitlab/cloud_pipelines/stacks/aws-e4s-aarch64/spack.yaml [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+    - wannier90
+    - zfp
+    #- dealii
+    #- geopm
+    #- qt
+    #- qwt
+    #- umpire # unsatisfiable concretization conflict w/ blt
+
+  - compiler:
+    - '%gcc'
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/aws-e4s-aarch64/spack.yaml]: Why not specify the compiler version here?  Maybe it's no big deal, just wondering.

File: share/spack/gitlab/cloud_pipelines/stacks/aws-e4s/spack.yaml [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+    - veloc
+    - wannier90
+    - zfp
+    #- dealii
+    #- geopm
+    #- qt
+    #- qwt
+
+  - compiler:
+    - '%gcc'
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/aws-e4s/spack.yaml]: Same question here.

**Review by scottwittenburg — CHANGES_REQUESTED** [review]
Thanks @bollig looks pretty good.  If you could just change the mirror urls to match the stacks (maybe one is already fine), then I think it could be merged.  Also I had a question about omitting the gcc version, same as for #30462.

File: share/spack/gitlab/cloud_pipelines/stacks/aws-e4s/spack.yaml (hunk 3 above)
Comment (bollig) [reply share/spack/gitlab/cloud_pipelines/stacks/aws-e4s/spack.yaml]: updated

File: share/spack/gitlab/cloud_pipelines/stacks/aws-e4s-aarch64/spack.yaml (hunk 2 above)
Comment (bollig) [reply share/spack/gitlab/cloud_pipelines/stacks/aws-e4s-aarch64/spack.yaml]: updated

**Review by scottwittenburg — APPROVED** [review]

**Comment by alalazo** [comment]
Are we going to resubmit this shortly?

**Comment by scottwittenburg** [comment]
I was leaving it to @bollig, but maybe I should just use the button in github to revert the revert PR?

**Comment by scottwittenburg** [comment]
What is the status here?

Should I take this over and work in the changes from #30753?  @becker33, could this still be backported to `v0.18` tomorrow if we got it merged to `develop`?

**Comment by bollig** [comment]
This shows merged from my view, and I dont have an option to revert the revert. Would appreciate your help if you know how to get the changes back in and update for 30753

**Pull request merged**